src/lib/datepicker/datepicker-toggle.ts

```diff
@@ -25,6 +27,13 @@ import {MatDatepicker} from './datepicker';
 import {MatDatepickerIntl} from './datepicker-intl';
 
 
+/** Can be used to override the icon of a `matDatepickerToggle`. */
+@Directive({
+  selector: '[matDatepickerToggleIcon]'
+})
+export class MatDatepickerToggleIcon {}
```

Comment:
Do we need a special directive? what if we just treat any projected content as a custom icon?

Reply:
We could, but it gets a little dirtier when you try to figure out whether to hide the default icon. We'd have to loop through all the `children` of the content element and count the amount of element nodes (skipping comments, text nodes etc.) on every change detection.

Reply:
Interesting, I thought there was an easier way to check if any content was projected, but it looks like you're right